adbms/src/main/scala/de/up/hpi/informationsystems/adbms/function/SequentialFunctor.scala

```diff
@@ -152,21 +156,56 @@ object SequentialFunctor {
+      *
+      * Calling this method multiple times will result in a list of sequential steps.
+      * <em>Call Order Matters!</em>
+      *
+      * @param transform  mapping function, transforming the received result with the functor context into a new request message
+      * @param recipients recipient list
+      * @tparam N         message type of the next request
+      * @return           a new [[de.up.hpi.informationsystems.adbms.function.SequentialFunctor.SeqFunctorBuilder]]
+      */
+    def nextWithContext[N <: Message]
+                       (transform: (Success[M], FunctorContext[S]) => Request[N], recipients: Seq[ActorSelection]): SeqFunctorBuilderStarted[S, N] =
+      new SeqFunctorBuilderStarted[S, N](start, steps :+ IntermediateStep(transform, recipients).toUpperBound)
 
     /** Concludes the definition of the sequential functor with a last transformation of the result message.
       *
-      * Use `end(identity)` to send the same result message back to the creator of the sequential functor.
+      * Use `endIdentity` to send the same result message back to the creator of the sequential functor.
       *
       * @param end mapping function, transforming the result message to another format
       * @tparam E  message type of the last success message, received by the creator of the sequential functor
       *            containing the result
       * @return    the sequential functor definition
       */
     def end[E <: Message]
-           (end: Success[M] => Success[E]): SequentialFunctorDef[A, Success[E]] =
-      new SequentialFunctorDef[A, Success[E]](start, steps, EndStep(end))
+           (end: Success[M] => Success[E]): SequentialFunctorDef[S, Success[E]] =
+      new SequentialFunctorDef[S, Success[E]](start, steps, EndStep( (e: Success[M], _: FunctorContext[S]) => end(e) ))
+
+    /** Concludes the definition of the sequential functor with a last transformation of the result message
+      * integrating the start message into the `end` function.
+      *
+      * Use `endIdentity` to send the same result message back to the creator of the sequential functor.
+      *
+      * @param end mapping function, transforming the result message to another format
+      * @tparam E  message type of the last success message, received by the creator of the sequential functor
+      *            containing the result
+      * @return    the sequential functor definition
+      */
+    def endWithContext[E <: Message]
+                      (end: (Success[M], FunctorContext[S]) => Success[E]): SequentialFunctorDef[S, Success[E]] =
+      new SequentialFunctorDef[S, Success[E]](start, steps, EndStep(end))
+
+    /** Concludes the definition of the sequential functor with sending the last received message back to the creator
+      * of the sequential functor.
+      *
+      * @return the sequential functor definition
+      */
+    def endIdentity: SequentialFunctorDef[S, Success[M]] = end(identity[Success[M]])
```

Comment:
This is the only change I am unsure of, because I think it clutters the interface and obfuscates the way the interface works.